frontend/src/components/pages/CreateGame/views/ChooseCategories.js

```diff
@@ -1,20 +1,197 @@
 
 
 export default class ChooseCategories extends React.Component {
 
+    static TITLE = 'Choisissez des thèmes';
+
+    pickedCategories = [];
+
     constructor(props) {
         super(props);
+        this.state = {
+            gameConfiguration: Util.getObjectFromSessionStorage(GameUtil.GAME_CONFIGURATION.key),
+            isLoading: true,
+            categories: false,
+            pickAllDisabled: false,
+            unpickAllDisabled: true,
+            nextButtonDisabled: true,
+        }
     }
 
     componentDidMount() {
-        console.log("la conf de la game", Util.getObjectFromSessionStorage('gameConfiguration'))
-        GameUtil.checkGameConfiguration(this.props.history);
+        (async () => {
+            try {
+                console.log("la conf de la game", Util.getObjectFromSessionStorage('gameConfiguration'));
+                const checkConfiguration = GameUtil.checkGameConfiguration(this.props.history);
+
+                if (!checkConfiguration.verified) {
+                    this.props.history.replace(checkConfiguration.redirect);
+
+                } else {
+                    const gameConfiguration = Util.getObjectFromSessionStorage(GameUtil.GAME_CONFIGURATION.key);
+                    const categories = await this.getCategories();
+
+                    this.setState({
+                        isLoading: false,
+                        categories
+                    });
+
+                    if (gameConfiguration.categories.length > 0) {
+                        gameConfiguration.categories.map(category => this.pickCategory(category));
+                    }
+                }
+            } catch (error) {
+                console.log(error)
+            }
+        })();
     }
 
+    getCategories = async () => {
+        const response = await Util.performAPIRequest('/game/categories');
+
+        if (!response.ok) throw 'error';
+
+        const responseData = await response.json();
+
+        return responseData.categories;
+    };
+
+    pickCategory = (category) => {
+
+        if (!this.pickedCategories.includes(category)) {
+            this.pickedCategories.push(category)
+
+        } else {
+            this.pickedCategories.splice(this.pickedCategories.indexOf(category), 1);
+        }
+
+        const categoryElt = document.querySelector(`#category-${category.name.toLowerCase()}`);
+        categoryElt.classList.toggle('purple-heart-bg');
+
+        console.log("Categories choisies", this.pickedCategories);
+        this.handleButtonsState();
+    };
+
+    pickAll = () => {
+        const { categories } = this.state;
+
+        this.pickedCategories = [];
+
+        this.pickedCategories.push(...categories);
+
+        const categoriesElt = Array.from(document.querySelectorAll('.category'));
+
+        categoriesElt.map((categoryElt) => {
+            categoryElt.classList.remove('purple-heart-bg');
+            categoryElt.classList.add('purple-heart-bg');
```

Comment:
Maybe here we should use a class with a more descriptive name, like "selected" or "active"